Client/game_sa/CEntitySA.cpp

```diff
@@ -606,101 +606,95 @@ bool CEntitySA::GetBoneRotation(eBone boneId, float& yaw, float& pitch, float& r
-        if (frameData)
-        {
-            RtQuat* boneOrientation = &frameData->m_pIFrame->orientation;
-            RwV3d   angles = {yaw, roll, pitch};
-            BoneNode_cSAInterface::EulerToQuat(&angles, boneOrientation);
-            CEntitySAInterface* theInterface = GetInterface();
-            if (theInterface)
-            {
-                theInterface->bDontUpdateHierarchy = false;
-            }
-            return true;
-        }
-    }
-    return false;
+    if (!clump)
+        return false;
+        
+    // updating the bone frame orientation will also update its children
+    // This rotation is only applied when UpdateElementRpHAnim is called
+    auto* clumpDataInterface = *pGame->GetClumpData(clump);
+    auto* frameData = clumpDataInterface->GetFrameDataByNodeId(boneId);
+    if (!frameData)
+        return false;
+        
+    RtQuat* boneOrientation = &frameData->m_pIFrame->orientation;
+    RwV3d angles = { yaw, roll, pitch };
+    BoneNode_cSAInterface::EulerToQuat(&angles, boneOrientation);
+    CEntitySAInterface* theInterface = GetInterface();
+    if (theInterface)
+        theInterface->bDontUpdateHierarchy = false;
+        
+    return true;
 }
 
 bool CEntitySA::SetBoneRotationQuat(eBone boneId, float x, float y, float z, float w)
 {
     RpClump* clump = GetRpClump();
-    if (clump)
-    {
-        // updating the bone frame orientation will also update its children
-        // This rotation is only applied when UpdateElementRpHAnim is called
-        CAnimBlendClumpDataSAInterface* clumpDataInterface = *pGame->GetClumpData(clump);
-        AnimBlendFrameData*             frameData = clumpDataInterface->GetFrameDataByNodeId(boneId);
-        if (frameData)
-        {
-            RtQuat* boneOrientation = &frameData->m_pIFrame->orientation;
-            boneOrientation->imag.x = x;
-            boneOrientation->imag.y = y;
-            boneOrientation->imag.z = z;
-            boneOrientation->real = w;
-            CEntitySAInterface* theInterface = GetInterface();
-            if (theInterface)
-            {
-                theInterface->bDontUpdateHierarchy = false;
-            }
-            return true;
-        }
-    }
+    if (!clump)
+        return false;
+        
+    // updating the bone frame orientation will also update its children
+    // This rotation is only applied when UpdateElementRpHAnim is called
+    auto* clumpDataInterface = *pGame->GetClumpData(clump);
+    auto* frameData = clumpDataInterface->GetFrameDataByNodeId(boneId);
+    if (!frameData)
+        return false;
+
+    RtQuat* boneOrientation = &frameData->m_pIFrame->orientation;
+    boneOrientation->imag.x = x;
+    boneOrientation->imag.y = y;
+    boneOrientation->imag.z = z;
+    boneOrientation->real = w;
+    CEntitySAInterface* theInterface = GetInterface();
+    if (theInterface)
+        theInterface->bDontUpdateHierarchy = false;
+        
+    return true;
     return false;
```

Comment:
Mistake

Reply:
Turns out that my brain is still asleep. Completely missed that in my review

Reply:
Yeah, I noticed it after merging.